types/htmlparser2/index.d.ts

```diff
@@ -1,30 +1,20 @@
-// Type definitions for htmlparser2 v3.7.x
+// Type definitions for htmlparser2 v3.10.x
 // Project: https://github.com/fb55/htmlparser2/
 // Definitions by: James Roland Cabresos <https://github.com/staticfunction>
 //                 Linus Unnebäck <https://github.com/LinusU>
+//                 Johan Davidsson <https://github.com/johandavidson>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 ///<reference types="node"/>
+///<reference types="domhandler"/>
+///<reference types="domutils"/>
 
 import { Writable } from 'stream'
+import { DomHandler } from 'domhandler';
+import * as DomUtils from 'domutils';
+export { DomElement, DomHandlerOptions, DomHandler, Element, Node } from 'domhandler';
 
-export interface Handler {
-    onopentag?: (name: string, attribs: { [type: string]: string }) => void;
-    onopentagname?: (name: string) => void;
-    onattribute?: (name: string, value: string) => void;
-    ontext?: (text: string) => void;
-    onclosetag?: (text: string) => void;
-    onprocessinginstruction?: (name: string, data: string) => void;
-    oncomment?: (data: string) => void;
-    oncommentend?: () => void;
-    oncdatastart?: () => void;
-    oncdataend?: () => void;
-    onerror?: (error: Error) => void;
-    onreset?: () => void;
-    onend?: () => void;
```

Comment:
The `DomHandler` interface has all function as mandatory, whereas the previous `Handler` interface had each function as optional. I think that they are optional in the library, so we should probably reflect that in the typings...

Reply:
No, those functions are added to `DomHandler` in the code implementation (just as they are in the `Parser`, see for example the `Parser`s `write` method). They are not optional, they are there. Maybe the old `Handler` interface reflected how `htmlparser2` was implemented before the handler was extracted to its own module?